File: apps/pwa/src/server/search_music.ts
import { MusicType } from '#/constants/music';
import { request } from '.';

function searchMusic({
  keyword,
  page,
  pageSize,
}: {
  keyword: string;
  page: number;
  pageSize: number;
}) {
  return request<{
    total: number;
    musicList: {
      id: string;
      type: MusicType;
      name: string;
      aliases: string[];
      cover: string;
      sq: string;
      hq: string;
      ac: string;
      singers: {
        id: string;
        avatar: string;
        name: string;
        aliases: string[];
      }[];
      createUser: {
        id: string;
        nickname: string;
        avatar: string;
      };
    }[];
  }>({
    path: '/api/music/search',
    params: { keyword, page, pageSize },
    withToken: true,
  });
}

export default searchMusic;

File: apps/flutter/lib/server/api/search_music.dart
import '../../models/music.dart';
import '../../server/request.dart';

class SearchMusicResponse {
  final int total;
  final List<Music> musicList;

  SearchMusicResponse({required this.total, required this.musicList});

  factory SearchMusicResponse.fromJson(Map<String, dynamic> json) =>
      SearchMusicResponse(
        total: json['total'],
        musicList: (json['musicList'] as List)
            .map((e) => Music.fromJson(e))
            .toList(),
      );
}

Future<SearchMusicResponse> searchMusic({
  required String keyword,
  int page = 1,
  int pageSize = 50,
}) async {
  final response = await httpGet(
    path: '/api/music/search',
    query: {
      'keyword': keyword,
      'page': page.toString(),
      'pageSize': pageSize.toString(),
    },
    withToken: true,
  );
  return SearchMusicResponse.fromJson(response);
}
